packages/core/src/client/client.ts

```diff
@@ -259,7 +259,38 @@ export abstract class Client {
     limit?: NumLike,
     after?: string,
   ): Promise<ClientFindCellsResponse> {
-    const res = await this.findCellsPagedNoCache(key, order, limit, after);
+    const myKey = ClientIndexerSearchKey.from(key);
+    const res = await this.findCellsPagedNoCache(myKey, order, limit, after);
+
+    // If no cells are returned, exit early.
+    if (res.cells.length === 0) {
+      return res;
+    }
+
+    // Assign type and lock scripts based on scriptType.
+    let type = myKey.filter?.script;
+    let lock: typeof type = myKey.script;
+    if (myKey.scriptType === "type") {
+      [lock, type] = [type, lock];
+    }
+
+    // Iterate through the returned cells to intern scripts hex fields from the source.
+    for (const { cellOutput } of res.cells) {
+      for (const [destination, source] of [
+        // Iterate over lock and type
+        [cellOutput.lock, lock],
+        [cellOutput.type, type],
+      ]) {
+        // Intern the hex fields of the Scripts for instant future equality.
+        if (source && destination?.eq(source)) {
+          destination.codeHash = source.codeHash;
+          destination.hashType = source.hashType;
+          destination.args = source.args;
```

Comment:
This could cause a problem: the search mode can be in Prefix/Partial, making the result different from our parameters. You can leave the thing as it is, since `hexFrom` already did it for us.

Reply:
> This could cause a problem

Not a problem per se, cause we are checking for equality, so either it is equal or not.

We could restrict for **exact only**, but at that point either:
1. We throw an error when it's not equal, which would hamper your code coverage
2. We blindly substitute the script, major error if ever something goes wrong upstream

So what would work and waste the minimal amount of cycles?

**Interleaving check and assignment field by field, this way we dedupe string also in prefix searches.**

I'm gonna go ahead with this and you can review again

Reply:
Sorry for the mistake before. You're right, this shouldn't cause a problem because of the equality check.
But why would we need this code? Seems like it does nothing.

If the "dedupe string" you're referring to is using the same string instance for scripts to reduce memory usage, I don't think that's what a `Client` should do. It's hardly possible; `hexFrom` should handle this; The V8 engine might be the place for this. 

This interface only returns ten cells by default each time, and I'm unsure if this will result in a noticeable performance improvement. I would suggest benchmarking to see if this code is necessary.

Reply:
<img width="1562" height="1054" alt="image" src="https://github.com/user-attachments/assets/0286712a-7104-4954-b5a2-c6f37bbd1159" />
The V8 engine handles a vast number of strings daily. It's better to let them optimize this rather than we do it manually.

Reply:
I'll write the benchmark, not sure how due to the deduping at compilation time, which does not apply here.

I'll think about it

Reply:
Benchmark at hand, you were right